tinygrad/codegen/lowerer.py

```diff
@@ -58,8 +70,15 @@ def lower(self, ast:UOp, opts:Renderer) -> UOp:
         self.idxs = get_grouped_dims("idx", full_shape[:global_dims], opts.global_max, reverse=True)
       else:
         # define indexes for GPU-like execution
-        self.idxs = get_grouped_dims("gidx", full_shape[:global_dims], opts.global_max, reverse=True) + \
-                    get_grouped_dims("lidx", full_shape[global_dims:first_reduce+group_for_reduces], opts.local_max)
+        reduceops = [x for x in ast.parents if x.op is UOps.REDUCE_AXIS and all(i < first_reduce+group_for_reduces for i in x.arg[1])]
```

Comment:
this is `i < first_reduce+group_for_reduces` instead of `i >= first reduce and i < first_reduce+group_for_reduces` because there shouldn't be a `UOps.REDUCE_AXIS` on a dim `< first_reduce `